pkg/reconciler/managed/reconciler.go

```diff
@@ -676,13 +676,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	)
 
 	// Check the pause annotation and return if it has the value "true"
-	// after logging, publishing an event and updating the SYNC status condition
-	if meta.IsPaused(managed) {
-		log.Debug("Reconciliation is paused via the pause annotation", "annotation", meta.AnnotationKeyReconciliationPaused, "value", "true")
-		record.Event(managed, event.Normal(reasonReconciliationPaused, "Reconciliation is paused via the pause annotation"))
+	// or if the management policies are enabled and the ManagementPolicy is empty.
+	// Log, publish an event and update the SYNC status condition.
+	if IsPaused(managed, r.managementPoliciesEnabled) {
```

Comment:
I know you didn't introduce this field, but wonder whether it would be better to just embed `feature.Flags` in `Reconciler` rather than using a `bool` field per flag. i.e.:

```go
type Reconciler struct {
	// Various fields redacted...

	// Replaces managementPoliciesEnabled
	features feature.Flags
}
```

This would:

* Make it easier to grep the codebase for the pattern - e.g. `r.features.Enabled(feature.SomeCoolFeature)`
* Make it easier to handle multiple feature flags in this reconciler in future.